### PR TITLE
Increase the kernel flash space for Raspberry Pi Pico

### LINKED ISSUE #3313: raspberry_pi_pico debug build fails with rust-lld error
When i  build TockOS to raspberry_pi_pico with make it builds normally, but when i try to build with make V=1 debug i have some rust-lld errors like this:
rust-lld: error: section '.text' will not fit in region 'rom': overflowed by 3 bytes

the entire log its in the link bellow
https://pastebin.com/ecnrHtq5

### PR DESCRIPTION
**Pull request by alexandruradovici**
### Pull Request Overview

This pull request fixes the Raspberry Pi Pico and similar boards kernel flash space. As several other peripherals were implemented, the kernel did not fit anymore.

This fixes #3313.

### Testing Strategy

N/A

### TODO or Help Wanted

N/A

### Documentation Updated

- [x] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [x] Ran `make prepush`.

### CODE REVIEW

File: boards/nano_rp2040_connect/layout.ld [hunk 1]
```diff
@@ -2,13 +2,13 @@ MEMORY
 {
   /* uncomment this to boot from RAM */
   /* reset (rx)  : ORIGIN = 0x20000000, LENGTH = 16K
-  rom (rx)  : ORIGIN = 0x20000100, LENGTH = 128K
+  rom (rx)  : ORIGIN = 0x20000100, LENGTH = 256K
```
Comment (hudson-ayers) [inline boards/nano_rp2040_connect/layout.ld:5]: Wouldn't the address and length of the next section need to be changed as well?

Comment (alexandruradovici) [reply boards/nano_rp2040_connect/layout.ld:5]: I think it already was in the right position before, 0x40000 should be 256K.

Comment (hudson-ayers) [reply boards/nano_rp2040_connect/layout.ld:5]: I see, thanks

**Review by hudson-ayers — APPROVED** [review]

**Review by twilfredo — APPROVED** [review]

**Review by alistair23 — APPROVED** [review]

**Pull request merged**